### PR TITLE
roachprod: fix decision of when to pass `cluster` parameter

### PR DESCRIPTION
**Pull request by renatolabs**
With this change, we should now support both shared-process multi-tenancy, external process multi-tenancy, pre-UA deployments, and shared-process deployments that set the default tenant to the application tenant.

Epic: none

Release note: None

### CODE REVIEW

**Comment by renatolabs** [comment]
#123765 introduced a bug when passing the `--cluster system` flag for clusters with shared-process MT. I forgot that the system tenant has service mode `external` as of #122276.

I've verified that this works in:

* pre-UA deployment
* shared-process: with and without a switch of the default controller
* external mode: we never want to pass `cluster` parameter unless it's the system tenant.

**Review by herkolategan — APPROVED** [review]

**Comment by renatolabs** [comment]
TFTR!

bors r=herkolategan

File: pkg/roachprod/install/cockroach.go [hunk 1]
```diff
@@ -623,9 +623,25 @@ func (c *SyncedCluster) NodeURL(
...
+	// The rules for when to include the `cluster` connection parameter
+	// are a little nuanced:
+	//
+	// Firstly, we only want to pass an explicit `cluster` name if the
+	// user provided one.
+	if virtualClusterName != "" &&
+		// If this is a shared service, we should always be passing the
+		// cluster connection parameter.
+		((serviceMode == ServiceModeShared) ||
+			// However, if this is an external process, this means that the
```
[the remark this answers is not in the document]
Comment (renatolabs) [reply pkg/roachprod/install/cockroach.go:635]: 1e1000 %

**Comment by renatolabs** [comment]
Was seeing some weird errors on my local shared-process branch and thought this had something to do with it. Turns out it was something else, merging again.

bors r=herkolategan

**Pull request merged**